src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -29,69 +37,124 @@ tags:
 
-![A Multi page form with a progress bar at the top.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600197013/progress_wl7bgp.png)
+Filling long forms can be stressful. This is largely true for almost everyone but perculiarly for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isin't what anyone wants. A simple solution is to break them into smaller sections. 
+This helps make long forms less dauting and easier to understand as it is provided in bits.
+
+The following principles apply for multi-step forms:
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
 
 
 ## Provide error messages if any
-If your site has forms, validation errors are inevitable. Always provide error messages when they occur. It is common practice to change the `<input>` color to red which is not ideal. Error messages helps people know exactly what they need to submit the form. 
 
-![A form with two fields indicating an error.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600197951/err_qiuycf.jpg)
+If your site has forms, validation errors are inevitable.
+When people encounter errors, clearly they need more help. Some people, like the visually impaired, may not even be aware that an error has occurred. It is common practice to change `<input>` color to red which is not ideal. This is because assistive devices will not catch the error, hence not informing the person of the error.
 
-## Avoid horizontal layout forms unless neccessary
-If possible, avoid making your form inputs next to each other. This allows people to follow the path to completion easily and focus on entering one piece of data at a time. 
 
-![A form with four input fields in a horizontal layout.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600201124/layout_glyubv.jpg)
+The following are ways to effectively and accessibly communicate form errors:
+- Let the person know there is an error.
+- Let the person access the error, and provide instructions on how to fix it.
+- Allow the person re-submit the form.
 
-If you must use a horizontal layout, make sure the tab order goes from left to right. This is done by adding a  `tabindex` attribute to `<input>`. The `tabindex` attribute should be in ascending order(1,2,3 and so on).
+
+In the code below, there is an empty [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) which report errors to assistive devices. Additionally, each field is followed by a _hint_ which is hidden by default but shown if that field does not validate.
 
 ```html
+<form>
+  <fieldset>

```

Comment:
why do we need a fieldset/legend to wrap the contents of the form?  this seems unnecessary for such a basic form.